src/java/za/co/mip/ablduck/utilities/CommentParser.java

```diff
@@ -0,0 +1,239 @@
+/**
+s * Copyright 2017 MIP Holdings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package za.co.mip.ablduck.utilities;
+
+import org.apache.tools.ant.Task;
+import org.markdown4j.Markdown4jProcessor;
+
+import za.co.mip.ablduck.javadoc.Javadoc;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+public class CommentParser {
+    private String comment;
+    private String source;
+    private HashMap<String, String> nestedComments = new HashMap<>();
+    private Javadoc javadocParser;
+    
+    public CommentParser (Task ablduck) {
+        javadocParser = new Javadoc(ablduck);
+    }
+    
+    public String markdown(String comment) throws IOException{
+        String markdown = "";
+
+        Markdown4jProcessor processor = new Markdown4jProcessor();
+        markdown = processor.process(comment);
+
+        return markdown;
+    }
+    
+    public void trimCommentLines(){
+        
+        Integer start = comment.indexOf("\n");

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Put single-quotes around '\n' to use the faster "indexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS3027)